client/lib/domains/get-wpcom-domain.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * Internal dependencies
+ */
+import { type as domainTypes } from './constants';
+
+export function getWpcomDomain( domainObjects ) {
+	const wpcomDomainObjects = domainObjects.filter( isWpcomDomain );
+
+	if ( wpcomDomainObjects.length === 0 ) {
+		return null;
+	}
+
+	// If there are .wpcomstaging and .wordpress domains, give preference to the .wpcomstaging one
+	const wpcomStagingDomainObject = wpcomDomainObjects.filter( isWpcomStagingDomain );
+	if ( wpcomStagingDomainObject.length > 0 ) {
+		return wpcomStagingDomainObject[ 0 ];
+	}
+
+	return wpcomDomainObjects[ 0 ];
+}
+
+function isWpcomDomain( domainObject ) {
+	return domainObject.type === domainTypes.WPCOM;
+}
+
+function isWpcomStagingDomain( domainObject ) {
+	return domainObject.domain.endsWith( '.wpcomstaging.com' );
```

Comment:
To be on the safe side and to simplify code a bit, you could write this as:

```suggestion
function isWpcomStagingDomain( { domain = '' } ) {
	return domain.endsWith( '.wpcomstaging.com' );
```

Reply:
Unfortunately it seems it is not valid JS indeed 😞 but I implemented the suggested modification 👍